tests/integration/test_package_managers.py

```diff
@@ -161,3 +182,24 @@ def test_packages(
         )
         log.info("Compare checksums of files in source vendor folder")
         assert files_checksums == expected_files_checksums
+
+    if test_params.build_image_with_fetched_deps:
+        log.info("Create cachi2.env file")
+        env_vars_file = os.path.join(tmpdir, "cachi2.env")
+        cmd = ["generate-env", output_folder, "--output", env_vars_file, "--for-output-dir", "/tmp"]
+        (output, rc) = cachi2_image.run_cmd_on_image(cmd, tmpdir)
+        assert rc == 0, f"Env var file creation failed. output-cmd: {output}"
+
+        log.info("Build container image with all prerequisites retrieved in previous steps")
+        container_folder = os.path.join(test_data_dir, test_case, "container")
+        test_image = utils.build_image(tmpdir, container_folder, test_case)
+
+        log.info(
+            f"Run installed app {test_params.install_app} "
+            f"on built image {test_image.repository}"
+        )
+        (output, rc) = test_image.run_cmd_on_image([test_params.install_app, "--help"], tmpdir)
+        assert rc == 0, (
+            f"{test_params.install_app} was not installed on image." f"output-cmd: {output}"
+        )
+        test_image.delete_image()
```

Comment:
If the assertion fails, the image will not be deleted. Maybe ContainerImage could be a context manager and delete the image on `__exit__`?

```python
with utils.build_image(...) as test_image:
    assert ...
```
Or a `try: assert ... finally: delete` would also work

Reply:
Great idea, thank you.  :taco:   Changed, please check. :)